server/util.go

```diff
@@ -197,9 +199,39 @@ func CreateSDKLogger(ctx *Context, out io.Writer) (log.Logger, error) {
 	// Check if the CometBFT flag for trace logging is set and enable stack traces if so.
 	opts = append(opts, log.TraceOption(ctx.Viper.GetBool("trace"))) // cmtcli.TraceFlag
 
+	// Error fields should be set under error object
+	zerolog.ErrorFieldName = "error"
+
+	// Add the kind and message field
+	zerolog.ErrorMarshalFunc = func(err error) interface{} {
+		stackArr, ok := pkgerrors.MarshalStack(errorspkg.WithStack(err)).([]map[string]string)
+		if !ok {
+			return struct{}{}
+		}
+		objectToReturn := DatadogErrorTrackingObject{
+			// Discard common prefix stack traces from zerolog call sites
+			Stack:   stackArr[5:],
```

Comment:
We should have a test for this to ensure that we are discarding exactly how much we want in case the stack depth changes due to a change in the implementation of this code or the underlying library.